lib/internal/test_runner/yaml_parser.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+const {
+  codes: {
+    ERR_TEST_FAILURE,
+  }
+} = require('internal/errors');
+const AssertionError = require('internal/assert/assertion_error');
+const {
+  ArrayPrototypeJoin,
+  Error,
+  Number,
+  NumberIsNaN,
+  RegExpPrototypeExec,
+  StringPrototypeEndsWith,
+  StringPrototypeRepeat,
+  StringPrototypeSlice,
+  StringPrototypeStartsWith,
+  StringPrototypeSubstring,
+} = primordials;
+
+const kYamlKeyRegex = /^(\s+)?(\w+):(\s)+([>|][-+])?(.*)$/;
+const kStackDelimiter = '    at ';
+
+function reConstructError(parsedYaml) {
+  if (!('error' in parsedYaml)) {
+    return parsedYaml;
+  }
+  const isAssertionError = parsedYaml.code === 'ERR_ASSERTION' ||
+    'actual' in parsedYaml || 'expected' in parsedYaml || 'operator' in parsedYaml;
+  const isTestFailure = parsedYaml.code === 'ERR_TEST_FAILURE' || 'failureType' in parsedYaml;
+  const stack = parsedYaml.stack ? kStackDelimiter + ArrayPrototypeJoin(parsedYaml.stack, `\n${kStackDelimiter}`) : '';
+
+  let cause;
+  let error;
+
+  if (isAssertionError) {
+    cause = new AssertionError({
+      message: parsedYaml.error,
+      actual: parsedYaml.actual,
+      expected: parsedYaml.expected,
+      operator: parsedYaml.operator
+    });
+  } else {
+    // eslint-disable-next-line no-restricted-syntax
+    cause = new Error(parsedYaml.error);
+    cause.code = parsedYaml.code;
+  }
+  cause.stack = stack;
+
+  if (isTestFailure) {
+    error = new ERR_TEST_FAILURE(cause, parsedYaml.failureType);
+    error.stack = stack;
+  }
+
+  parsedYaml.error = error ?? cause;
+  delete parsedYaml.stack;
+  delete parsedYaml.code;
+  delete parsedYaml.failureType;
+  delete parsedYaml.actual;
+  delete parsedYaml.expected;
+  delete parsedYaml.operator;
+
+  return parsedYaml;
+}
+
+function getYamlValue(value, key) {
```

Comment:
`key` is unused, right?
```suggestion
function getYamlValue(value) {
```